### PR TITLE
Localize

### LINKED ISSUE #242: Localize Strings
add Localize functions ....

### PR DESCRIPTION
**Pull request by 6543**
close #242

[Crowdin](https://crowdin.com/project/cabr2)

### CODE REVIEW

**Comment by 6543** [comment]
@Epsilon02 would be nice if you could tell me at witch point I can read the set locale from user config ...

File: src/app/@core/services/i18n/i18n.service.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import { strings as en } from '../../../../assets/strings_en.json';
+
+export class i18n {
+  static getStrings(lang: string): any {
+    if (lang === 'de') {
+      return de;
+    }
+    return en;
+  }
+}
```
Comment (crapStone) [inline src/app/@core/services/i18n/i18n.service.ts:11]: I'll rewrite this as a angular service to get the data from the backend.

File: src/app/@core/services/i18n/i18n.service.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  getAvailableLanguages(): Observable<LocalizedStringsHeader[]> {
+    return this.tauriService.promisified({
+      cmd: 'getAvailableLanguages',
+    });
+  }
+
+  getLocalizedStrings(language: string): Observable<LocalizedStrings> {
+    logger.error('get strings', language);
```
Comment (Epsilon02) [inline src/app/@core/services/i18n/i18n.service.ts:31]: Why logging here an error?

**Review by Epsilon02 — APPROVED** [review]

File: src-tauri/cabr2_config/src/types.rs [hunk 3]
```diff
@@ -134,4 +142,16 @@ impl std::convert::From<JsonLogging> for TomlLogging {
 
 pub type GHSSymbols = HashMap<String, String>;
 
+#[derive(Debug, Deserialize, Serialize)]
+pub struct LocalizedStringsHeader {
+  version: u8,
```
Comment (6543) [inline src-tauri/cabr2_config/src/types.rs:147]: rm

File: src/app/app.component.ts [hunk 4]
```diff
@@ -19,34 +19,49 @@ const logger = new Logger('main');
...
     private renderer: Renderer2,
+
     private config: ConfigModel,
     private global: GlobalModel,
     private configService: ConfigService,
     private alertService: AlertService,
+    private i18nService: I18nService,
   ) {
+    this.global.localizedStringsObservable.subscribe((strings) => this.strings = strings);
+    this.i18nService.getAvailableLanguages().subscribe(
```
Comment (crapStone) [inline src/app/app.component.ts:35]:
```suggestion
    // this is for testing only
    this.i18nService.getAvailableLanguages().subscribe(
```

**Review by crapStone — APPROVED** [review]

**Review by 6543 — COMMENTED** [review]
LGTM

**Review by crapStone — APPROVED** [review]

**Pull request merged**